hashes/src/sha256.rs

```diff
@@ -114,33 +120,41 @@ impl Hash {
     /// Computes hash from `bytes` in `const` context.
     ///
     /// Warning: this function is inefficient. It should be only used in `const` context.
-    pub const fn const_hash(bytes: &[u8]) -> Self { Hash(Midstate::const_hash(bytes, true).0) }
+    pub const fn const_hash(bytes: &[u8]) -> Self { Hash(Midstate::const_hash(bytes, true).bytes) }
 }
 
 /// Output of the SHA256 hash function.
 #[derive(Copy, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Hash)]
```

Comment:
Nope, We can't have both comparison traits and `Borrow` since they don't act the same. I think we should just remove `Borrow`.

Reply:
Oh of course, because borrow returns just the bytes and equivalence uses the whole struct. Remove `Borrow` and keep `AsRef`, right?

Reply:
Yes.